File: src/core/exceptions.py
"""
Custom exceptions for Docker Compose Manager.
"""


class DockerComposeError(Exception):
    """Base exception for Docker Compose Manager errors."""
    pass


class ConfigError(DockerComposeError):
    """Raised when there's an error with configuration loading or validation."""
    pass


class DeploymentError(DockerComposeError):
    """Raised when deployment operations fail."""
    pass


class MonitoringError(DockerComposeError):
    """Raised when monitoring operations fail."""
    pass


class BackupError(DockerComposeError):
    """Raised when backup/restore operations fail."""
    pass


class ValidationError(DockerComposeError):
    """Raised when configuration validation fails."""
    pass


class EnvironmentError(DockerComposeError):
    """Raised when environment-related operations fail."""
    pass

File: src/implementations/javascript/exceptions.js
/**
 * Custom exceptions for JavaScript implementation
 */

class DockerComposeError extends Error {
  constructor(message) {
    super(message);
    this.name = 'DockerComposeError';
  }
}

class ConfigError extends DockerComposeError {
  constructor(message) {
    super(message);
    this.name = 'ConfigError';
  }
}

class DeploymentError extends DockerComposeError {
  constructor(message) {
    super(message);
    this.name = 'DeploymentError';
  }
}

class MonitoringError extends DockerComposeError {
  constructor(message) {
    super(message);
    this.name = 'MonitoringError';
  }
}

class BackupError extends DockerComposeError {
  constructor(message) {
    super(message);
    this.name = 'BackupError';
  }
}

class ValidationError extends DockerComposeError {
  constructor(message) {
    super(message);
    this.name = 'ValidationError';
  }
}

class EnvironmentError extends DockerComposeError {
  constructor(message) {
    super(message);
    this.name = 'EnvironmentError';
  }
}

module.exports = {
  DockerComposeError,
  ConfigError,
  DeploymentError,
  MonitoringError,
  BackupError,
  ValidationError,
  EnvironmentError
};
